cpp/CMakeLists.txt

```diff
@@ -447,7 +447,7 @@ if (UNIX)
   add_custom_target(lint ${CPPLINT_BIN}
   --verbose=2
   --linelength=90
-  --filter=-whitespace/comments,-readability/todo,-build/header_guard,-build/c++11,-runtime/references,-build/include_order
+  --filter=-whitespace/comments,-readability/todo,-build/header_guard,-build/c++11,-runtime/references,-build/include_order,-build/include_what_you_use
```

Comment:
Otherwise you get errors such as:
```
/home/antoine/arrow/cpp/src/plasma/plasma.h:123:  Add #include <memory> for unique_ptr<>  [build/include_what_you_use] [4]
```

Reply:
Ah, I think we do want these errors -- at some point we should be more rigorous about enforcing IWYU (https://github.com/apache/arrow/blob/master/cpp/README.md#keeping-includes-clean-with-include-what-you-use) but cpplint catches some of the more obvious issues, like `<vector>` and `<string>`. We can add the missing includes here

Reply:
I would also like to have them, they have been very helpful. @pitrou does the new version add a lot of errors? Then we would add and do a second PR that fixes them (happy to take care of that).

Reply:
There aren't too many errors, I'm fixing them now